### PR TITLE
Add Interfaces to maqs-java

### PR DESCRIPTION
**Pull request by jredingcsv**
added interfaces to base, appium, cucumber, selenium, logger, perf timer, and webservices.
Updated some comments
renamed ManagerDictionary to ManagerStore to match maqs-dotnet naming

### CODE REVIEW

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObjectUnitTest.java [hunk 1]
```diff
@@ -128,7 +127,7 @@ private Supplier<HttpClient> getHttpClientSupplier() {
    *
    * @return a web service driver.
    */
-  private WebServiceDriver getWebServiceDriver() {
+  private WebServiceDriver getWebServiceDriverWithURI() {
```
Comment (twalshcsv) [inline maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObjectUnitTest.java:130]: Why did we rename this?
I am fine with it changing this is wordier and the casing is funky.
If you are keeping this name change, may I suggest  getWebServiceDriverWith**Uri** instead?

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObjectUnitTest.java [hunk 2]
```diff
@@ -16,14 +15,14 @@
 /**
  * The Web Service Test Object unit test class.
  */
-public class WebServiceTestObjectUnitTest extends BaseGenericTest {
+public class WebServiceTestObjectUnitTest extends BaseWebServiceTest {
```
Comment (twalshcsv) [inline maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObjectUnitTest.java:18]: Suggest changing this back
These tests do not benefit from using the service base.
I would actually argue these tests are more valuable with base generic.

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverUnitTest.java [hunk 3]
```diff
@@ -12,7 +12,7 @@
 /**
  * The Web Service Driver unit test class.
  */
-public class WebServiceDriverUnitTest {
+public class WebServiceDriverUnitTest extends BaseWebServiceTest {
```
Comment (twalshcsv) [inline maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverUnitTest.java:15]: Why are we extending base service if it is not required for the tests?

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java [hunk 4]
```diff
@@ -22,7 +18,7 @@
 /**
  * The Web Driver Service Driver Put unit test class.
  */
-public class WebServiceDriverPutUnitTest {
+public class WebServiceDriverPutUnitTest extends BaseWebServiceTest {
```
Comment (twalshcsv) [inline maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java:21]: Why are we extending base service if it is not required for the tests?

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceConfigUnitTest.java [hunk 5]
```diff
@@ -5,15 +5,14 @@
...
-import com.cognizantsoftvision.maqs.base.BaseGenericTest;
 import com.cognizantsoftvision.maqs.utilities.helper.TestCategories;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
 /**
  * The Web Service Config unit test class.
  */
-public class WebServiceConfigUnitTest extends BaseGenericTest {
+public class WebServiceConfigUnitTest extends BaseWebServiceTest {
```
Comment (twalshcsv) [inline maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceConfigUnitTest.java:15]: Why are we extending base service if it is not required for the tests?

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/MediaTypeUnitTest.java [hunk 6]
```diff
@@ -4,14 +4,13 @@
...
 
-import com.cognizantsoftvision.maqs.base.BaseGenericTest;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
 /**
  * The Media Type unit test class.
  */
-public class MediaTypeUnitTest extends BaseGenericTest {
+public class MediaTypeUnitTest extends BaseWebServiceTest {
```
Comment (twalshcsv) [inline maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/MediaTypeUnitTest.java:13]: Why are we extending base service if it is not required for the tests?

File: maqs-webservices/src/main/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObject.java [hunk 7]
```diff
@@ -5,14 +5,15 @@
...
-import com.cognizantsoftvision.maqs.utilities.logging.Logger;
+import com.cognizantsoftvision.maqs.utilities.logging.ILogger;
 import java.net.http.HttpClient;
 import java.util.function.Supplier;
 
 /**
  * The Web Service Test Object class.
  */
-public class WebServiceTestObject extends BaseTestObject {
+public class WebServiceTestObject extends BaseTestObject implements IWebServiceTestObject {
```
Comment (twalshcsv) [inline maqs-webservices/src/main/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObject.java:15]: Why are we implementing the base service interface if it is not required for the tests?

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/MediaTypeUnitTest.java (hunk 6 above)
Comment (twalshcsv) [inline maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/MediaTypeUnitTest.java:13]: Why are we implementing the base service interface if it is not required for the tests?

File: maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IFileLogger.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+   * @return Returns the File Name.
+   */
+  String getFileName();
+
+  /**
+   * Logs the message.
+   * @param message The message text.
+   * @param args String format arguments
+   */
+  void logMessage(String message, Object... args);
```
Comment (twalshcsv) [inline maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IFileLogger.java:52]: Maybe my Java is just rusty, but I don't think we want the logMessage functions in this interface as they are already part of the parent interface.

File: maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IFileLogger.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+   * @return returns the file path
+   */
+  String getFilePath();
+
+  /**
+   * Gets the Message Type value.
+   *
+   * @return The Message Type.
+   */
+  MessageType getMessageType();
```
Comment (twalshcsv) [inline maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IFileLogger.java:24]: Why is this here and how is different from "getLoggingLevel" in the ILogger interface

File: maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IHtmlFileLogger.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2022 (C) Cognizant SoftVision, All rights Reserved
+ */
+
+package com.cognizantsoftvision.maqs.utilities.logging;
+
+/**
+ * The Html File Logger interface class.
+ */
+public interface IHtmlFileLogger extends AutoCloseable {
```
Comment (twalshcsv) [inline maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IHtmlFileLogger.java:10]: Weird name for this interface.
I know we use if for html, but it is more generic than that.
IAutoCloseableLogger seems like a better name

Comment (twalshcsv) [inline maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IHtmlFileLogger.java:10]: Why doesn't this also have ILogger  added like we have for the file logger interface?

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumUtilitiesUnitTest.java [hunk 11]
```diff
@@ -24,11 +17,17 @@
 import org.openqa.selenium.support.events.WebDriverListener;
 import org.testng.Assert;
 import org.testng.annotations.Test;
+import java.io.File;
```
Comment (twalshcsv) [inline maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumUtilitiesUnitTest.java:20]: Why we not placing these in alphabetical order?

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java [hunk 12]
```diff
@@ -4,11 +4,7 @@
 
 package com.cognizantsoftvision.maqs.webservices.WebServiceDriverRequestUnitTest;
 
-import com.cognizantsoftvision.maqs.webservices.HttpClientFactory;
-import com.cognizantsoftvision.maqs.webservices.MediaType;
-import com.cognizantsoftvision.maqs.webservices.WebServiceConfig;
-import com.cognizantsoftvision.maqs.webservices.WebServiceDriver;
-import com.cognizantsoftvision.maqs.webservices.WebServiceUtilities;
+import com.cognizantsoftvision.maqs.webservices.*;
```
Comment (twalshcsv) [inline maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java:7]: Why are we moving to the wild card pattern? - Not sure this is a good idea

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/pageModel/HeaderPageModel.java [hunk 13]
```diff
@@ -4,10 +4,7 @@
 
 package com.cognizantsoftvision.maqs.selenium.pageModel;
 
-import com.cognizantsoftvision.maqs.selenium.BaseSeleniumPageModel;
-import com.cognizantsoftvision.maqs.selenium.SeleniumConfig;
-import com.cognizantsoftvision.maqs.selenium.SeleniumTestObject;
-import com.cognizantsoftvision.maqs.selenium.UIWait;
+import com.cognizantsoftvision.maqs.selenium.*;
```
Comment (twalshcsv) [inline maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/pageModel/HeaderPageModel.java:7]: Why are we moving to the wild card pattern? - Not sure this is a good idea

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/pageModel/AutomationPageModel.java [hunk 14]
```diff
@@ -4,10 +4,7 @@
 
 package com.cognizantsoftvision.maqs.selenium.pageModel;
 
-import com.cognizantsoftvision.maqs.selenium.LazyWebElement;
-import com.cognizantsoftvision.maqs.selenium.SeleniumConfig;
-import com.cognizantsoftvision.maqs.selenium.SeleniumTestObject;
-import com.cognizantsoftvision.maqs.selenium.UIWait;
+import com.cognizantsoftvision.maqs.selenium.*;
```
Comment (twalshcsv) [inline maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/pageModel/AutomationPageModel.java:7]: Why are we moving to the wild card pattern? - Not sure this is a good idea

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumTestObjectUnitTest.java [hunk 15]
```diff
@@ -13,7 +12,7 @@
 /**
  * The type Selenium test object test.
  */
-public class SeleniumTestObjectUnitTest extends BaseGenericTest {
+public class SeleniumTestObjectUnitTest extends BaseSeleniumTest {
```
Comment (twalshcsv) [inline maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumTestObjectUnitTest.java:15]: Why add a specific base when generic is all we need?

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumDriverManagerUnitTest.java [hunk 16]
```diff
@@ -14,7 +13,7 @@
 /**
  * The type Selenium driver manager test.
  */
-public class SeleniumDriverManagerUnitTest extends BaseGenericTest {
+public class SeleniumDriverManagerUnitTest extends BaseSeleniumTest {
```
Comment (twalshcsv) [inline maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumDriverManagerUnitTest.java:16]: Why add a specific base when generic is all we need?

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumConfigUnitTest.java [hunk 17]
```diff
@@ -17,7 +16,7 @@
 /**
  * Selenium configuration tests.
  */
-public class SeleniumConfigUnitTest extends BaseGenericTest {
+public class SeleniumConfigUnitTest extends BaseSeleniumTest {
```
Comment (twalshcsv) [inline maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumConfigUnitTest.java:19]: Why add a specific base when generic is all we need?

File: maqs-base/src/test/java/com/cognizantsoftvision/maqs/base/BaseTestObjectUnitTest.java [hunk 18]
```diff
@@ -35,18 +35,17 @@ public void testBaseTestObject1() {
    */
   @Test(groups = TestCategories.FRAMEWORK)
   public void testBaseTestObject2() {
-    final BaseTestObject testObject = this.getTestObject();
+    final BaseTestObject testObject = (BaseTestObject) this.getTestObject();
```
Comment (twalshcsv) [inline maqs-base/src/test/java/com/cognizantsoftvision/maqs/base/BaseTestObjectUnitTest.java:38]: Why are we doing all these **BaseTestObject** casts?
As far as I can tell they are not needed.

File: maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/MobileDriverManagerUnitTest.java [hunk 19]
```diff
@@ -15,7 +13,7 @@
 /**
  * The type Mobile driver manager unit test.
  */
-public class MobileDriverManagerUnitTest extends BaseGenericTest {
+public class MobileDriverManagerUnitTest extends BaseAppiumTest {
```
Comment (twalshcsv) [inline maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/MobileDriverManagerUnitTest.java:16]: Why extend base Appium when generic works?

File: maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumUtilitiesUnitTest.java [hunk 20]
```diff
@@ -25,7 +24,7 @@
  * Appium Utilities Unit Test class.
  */
 @Ignore
-public class AppiumUtilitiesUnitTest extends BaseGenericTest {
```
Comment (twalshcsv) [inline maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumUtilitiesUnitTest.java:28]: Why extend base Appium when generic works?

File: maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumDriverFactoryUnitTest.java [hunk 21]
```diff
@@ -25,7 +24,7 @@
 /**
  * The type Appium driver factory test.
  */
-public class AppiumDriverFactoryUnitTest extends BaseGenericTest {
+public class AppiumDriverFactoryUnitTest extends BaseAppiumTest {
```
Comment (twalshcsv) [inline maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumDriverFactoryUnitTest.java:27]: Why extend base Appium when generic works?

File: maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumConfigUnitTest.java [hunk 22]
```diff
@@ -19,7 +18,7 @@
 /**
  * The type Appium config unit test.
  */
-public class AppiumConfigUnitTest extends BaseGenericTest {
```
Comment (twalshcsv) [inline maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumConfigUnitTest.java:22]: Why extend base Appium when generic works?

**Review by twalshcsv — CHANGES_REQUESTED** [review]
Overall like the change but see some small issues - See comments for issue details

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObjectUnitTest.java (hunk 1 above)
Comment (jredingcsv) [reply maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObjectUnitTest.java]: changed the name back, was resolved with reference changed back to generictest

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObjectUnitTest.java (hunk 2 above)
Comment (jredingcsv) [reply maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceTestObjectUnitTest.java]: changed back to baseGenericTest

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverUnitTest.java (hunk 3 above)
Comment (jredingcsv) [reply maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverUnitTest.java]: changed back to baseGenericTest

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java (hunk 4 above)
Comment (jredingcsv) [reply maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java]: changed back to baseGenericTest

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceConfigUnitTest.java (hunk 5 above)
Comment (jredingcsv) [reply maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceConfigUnitTest.java]: changed back to baseGenericTest

File: maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IFileLogger.java (hunk 9 above)
Comment (jredingcsv) [reply maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IFileLogger.java]: deleted this and changed getLoggingLevel to getMessageType

File: maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IHtmlFileLogger.java [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+
+  /**
+   * Logs a message to the log file.
+   *
+   * @param message the string message to be logged
+   * @param args the object in addition to the string message
+   */
+  void logMessage(String message, Object... args);
```
[the remark this answers is not in the document]
Comment (jredingcsv) [reply maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IHtmlFileLogger.java]: deleted logMessage methods in this interface

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumUtilitiesUnitTest.java (hunk 11 above)
Comment (jredingcsv) [reply maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumUtilitiesUnitTest.java]: alphabetized the imports

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/WebDriverFactoryUnitTest.java [hunk 24]
```diff
@@ -29,12 +28,12 @@
 /**
  * The WebDriverFactory test class.
  */
-public class WebDriverFactoryUnitTest extends BaseGenericTest {
+public class WebDriverFactoryUnitTest extends BaseSeleniumTest {
```
[the remark this answers is not in the document]
Comment (jredingcsv) [reply maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/WebDriverFactoryUnitTest.java]: changed back to generic test

File: maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java (hunk 12 above)
Comment (jredingcsv) [reply maqs-webservices/src/test/java/com/cognizantsoftvision/maqs/webservices/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java]: brought back all imports

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/pageModel/HeaderPageModel.java (hunk 13 above)
Comment (jredingcsv) [reply maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/pageModel/HeaderPageModel.java]: brought back all imports

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/pageModel/AutomationPageModel.java (hunk 14 above)
Comment (jredingcsv) [reply maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/pageModel/AutomationPageModel.java]: brought back all imports

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumTestObjectUnitTest.java (hunk 15 above)
Comment (jredingcsv) [reply maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumTestObjectUnitTest.java]: reverted back to baseGeneric

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumDriverManagerUnitTest.java (hunk 16 above)
Comment (jredingcsv) [reply maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumDriverManagerUnitTest.java]: reverted back to baseGeneric

File: maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumConfigUnitTest.java (hunk 17 above)
Comment (jredingcsv) [reply maqs-selenium/src/test/java/com/cognizantsoftvision/maqs/selenium/SeleniumConfigUnitTest.java]: reverted back to baseGeneric

File: maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/MobileDriverManagerUnitTest.java (hunk 19 above)
Comment (jredingcsv) [reply maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/MobileDriverManagerUnitTest.java]: reverted back to use baseGenericTest

File: maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumUtilitiesUnitTest.java (hunk 20 above)
Comment (jredingcsv) [reply maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumUtilitiesUnitTest.java]: reverted back to use baseGenericTest

File: maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumDriverFactoryUnitTest.java (hunk 21 above)
Comment (jredingcsv) [reply maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumDriverFactoryUnitTest.java]: reverted back to use baseGenericTest

File: maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumConfigUnitTest.java (hunk 22 above)
Comment (jredingcsv) [reply maqs-appium/src/test/java/com/cognizantsoftvision/maqs/appium/AppiumConfigUnitTest.java:22]: reverted back to use baseGenericTest

File: maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IHtmlFileLogger.java (hunk 10 above)
Comment (jredingcsv) [reply maqs-utilities/src/main/java/com/cognizantsoftvision/maqs/utilities/logging/IHtmlFileLogger.java]: deleted this interface as IFileLogger extension works

File: maqs-base/src/test/java/com/cognizantsoftvision/maqs/base/BaseTestObjectUnitTest.java (hunk 18 above)
Comment (jredingcsv) [reply maqs-base/src/test/java/com/cognizantsoftvision/maqs/base/BaseTestObjectUnitTest.java]: resolved casting issues

**Review by twalshcsv — APPROVED** [review]
